File: app/src/main/java/com/gnoemes/shikimoriapp/domain/search/SearchInteractor.java
package com.gnoemes.shikimoriapp.domain.search;

import com.gnoemes.shikimoriapp.entity.anime.domain.Anime;
import com.gnoemes.shikimoriapp.entity.search.domain.FilterItem;

import java.util.HashMap;
import java.util.List;

import io.reactivex.Single;

public interface SearchInteractor {

    /**
     * Load anime list with filters
     */
    Single<List<Anime>> loadListWithFilters(HashMap<String, List<FilterItem>> filters, int page, int limit);

    /**
     * Load anime list with default filters
     */
    Single<List<Anime>> loadList(int page, int limit);
}

File: app/src/main/java/com/gnoemes/shikimoriapp/domain/search/SearchInteractor.kt
package com.gnoemes.shikimoriapp.domain.search

import com.gnoemes.shikimoriapp.entity.anime.domain.Anime
import com.gnoemes.shikimoriapp.entity.app.data.AppConfig
import com.gnoemes.shikimoriapp.entity.manga.domain.Manga
import com.gnoemes.shikimoriapp.entity.roles.domain.Character
import com.gnoemes.shikimoriapp.entity.roles.domain.Person
import com.gnoemes.shikimoriapp.entity.search.domain.FilterItem
import io.reactivex.Single

interface SearchInteractor {

    fun loadAnimeList(page: Int, limit: Int = AppConfig.DEFAULT_LIMIT): Single<List<Anime>>

    fun loadMangaList(page: Int, limit: Int = AppConfig.DEFAULT_LIMIT): Single<List<Manga>>

    fun loadRanobeList(page: Int, limit: Int = AppConfig.DEFAULT_LIMIT): Single<List<Manga>>

    fun loadCharacterListWithFilters(filters: HashMap<String, List<FilterItem>>?, page: Int = 0, limit: Int = AppConfig.DEFAULT_LIMIT): Single<List<Character>>

    fun loadPersonListWithFilters(filters: HashMap<String, List<FilterItem>>?, page: Int = 0, limit: Int = AppConfig.DEFAULT_LIMIT): Single<List<Person>>

    fun loadAnimeListWithFilters(filters: HashMap<String, List<FilterItem>>?, page: Int, limit: Int = AppConfig.DEFAULT_LIMIT): Single<List<Anime>>

    fun loadMangaListWithFilters(filters: HashMap<String, List<FilterItem>>?, page: Int, limit: Int = AppConfig.DEFAULT_LIMIT): Single<List<Manga>>

    fun loadRanobeListWithFilters(filters: HashMap<String, List<FilterItem>>?, page: Int, limit: Int = AppConfig.DEFAULT_LIMIT): Single<List<Manga>>
}